igvc/src/speed_compare/main.cpp

```diff
@@ -0,0 +1,77 @@
+#include <igvc_msgs/velocity_pair.h>
+#include <ros/ros.h>
+#include <std_msgs/Bool.h>
+#include <chrono>
+#include <fstream>
+
+bool enabled = false;
+std::ofstream file;
+float target_left_vel;
+float target_right_vel;
+ros::Time start_time;
+
+void enabled_callback(const std_msgs::BoolConstPtr& msg)
+{
+  enabled = msg->data;
+  if (enabled)
+  {
+    ROS_INFO_STREAM("The robot is enabled. Data will be written");
+  }
+  else
+  {
+    ROS_INFO_STREAM("The robot is disabled. Data will not be written.");
+  }
+}
+
+void motors_callback(const igvc_msgs::velocity_pair& msg)
+{
+  if (enabled)
+  {
+    target_left_vel = msg.left_velocity;
+    target_right_vel = msg.right_velocity;
+  }
+  else
```

Comment:
I would remove the logging here since it is just a repeat of the above callback information as well